pkg/goDB/db_writer.go

```diff
@@ -146,22 +148,29 @@ func (w *DBWriter) Write(flowmap AggFlowMap, meta BlockMetadata, timestamp int64
 	return update, err
 }
 
-func dbData(iface string, timestamp int64, aggFlowMap AggFlowMap) ([ColIdxCount][]byte, InterfaceSummaryUpdate) {
+func dbData(iface string, timestamp int64, aggFlowMap *hashmap.AggFlowMap) ([ColIdxCount][]byte, InterfaceSummaryUpdate) {
 	var dbData [ColIdxCount][]byte
 	summUpdate := new(InterfaceSummaryUpdate)
 
-	list := aggFlowMap.Flatten().Sort()
+	v4List, v6List := aggFlowMap.Flatten()
+	v4List = v4List.Sort()
+	v6List = v6List.Sort()
 	for i := columnIndex(0); i < ColIdxAttributeCount; i++ {
-		dbData[i] = make([]byte, 0, columnSizeofs[i]*len(list))
+		columnSizeof := columnSizeofs[i]
+		if columnSizeof == ipSizeOf {
+			dbData[i] = make([]byte, 0, 4*len(v4List)+16*len(v6List))
+		} else {
+			dbData[i] = make([]byte, 0, columnSizeofs[i]*(len(v4List)+len(v6List)))
+		}
 	}
 
 	summUpdate.Timestamp = time.Unix(timestamp, 0)
 	summUpdate.Interface = iface
 
-	// loop through the flow map to extract the relevant
+	// loop through the v4 & v6 flow maps to extract the relevant
 	// values into database blocks.
 	var bytesRcvd, bytesSent, pktsRcvd, pktsSent []uint64
-	for _, flow := range list {
+	for _, flow := range v4List {
```

Comment:
We can reduce _El Code Duplicator_ in this case. In both cases you are looping over a `hashmap.List`. Why not just  write:

```
for _, list := range []hashmap.List{v4List, v6List} {
	for _, flow := range list {

		summUpdate.FlowCount++
		summUpdate.Traffic += flow.NBytesRcvd
		summUpdate.Traffic += flow.NBytesSent

		// counters
		bytesRcvd = append(bytesRcvd, flow.NBytesRcvd)
		bytesSent = append(bytesSent, flow.NBytesSent)
		pktsRcvd = append(pktsRcvd, flow.NPktsRcvd)
		pktsSent = append(pktsSent, flow.NPktsSent)

		// attributes
		dbData[DportColIdx] = append(dbData[DportColIdx], flow.GetDport()...)
		dbData[ProtoColIdx] = append(dbData[ProtoColIdx], flow.GetProto())
		dbData[SipColIdx] = append(dbData[SipColIdx], flow.GetSip()...)
		dbData[DipColIdx] = append(dbData[DipColIdx], flow.GetDip()...)
	}
}
```

Reply:
Jup, can do!